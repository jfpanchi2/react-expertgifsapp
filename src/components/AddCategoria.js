import React, { useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategoria = ({setCategorias}) => {
    const [inputValue, setinputValue] = useState('');
    const hanldeInputChange = (e) => {
        //console.log(e.target.value);
        setinputValue(e.target.value);
    } 
    
    const handleSubmit =(e) =>{
        e.preventDefault();
        if(inputValue.trim().length>2){
            setCategorias(categorias=>[inputValue,...categorias]);
        }
        setinputValue("");
        //console.log('Submit exitoso');
       
    }
    return (
        <form onSubmit={handleSubmit}>
            <h2>Buscar gifs</h2>
            <input 
            type="text" 
            value={inputValue}
            onChange={hanldeInputChange}></input>
        </form>
        
    )
}

AddCategoria.propTypes ={
    setCategorias: PropTypes.func.isRequired
}