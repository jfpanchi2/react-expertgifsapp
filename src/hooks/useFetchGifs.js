import {useState, useEffect}   from 'react';
import { getGifs } from '../helpers/getGifs';
export const useFetchGifs = (categoria) => {
    const [state, setstate] = useState({
        data: [],
        loading: true,
    });

    useEffect(()=>{
        getGifs(categoria)
        .then(imagenes=>{
                setstate({
                    data: imagenes,
                    loading: false,
                });
        })
    },[categoria])
    
    return state;
}
