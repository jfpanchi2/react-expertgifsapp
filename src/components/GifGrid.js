import React  from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
export const GifGrid = ({categoria}) => {
    /* const [imagen, setimagen] = useState([]); */
    /* const [count, setcount] = useState(0); */
    
    const {data:imagenes,loading} = useFetchGifs(categoria)
    
    /* useEffect(()=>{
        getGifs(categoria)
        .then(imagenes =>setimagen(imagenes));
    },[categoria]) */
    
    //getGifs();
    return (
        <div>
            <h3 className="animate__animated animate__fadeIn">{categoria}</h3>
            {loading && <h4 className="animate__animated animate__flash">Cargando...</h4>}
            <div className="card-grid">
            
                {
                        imagenes.map(img =>{
                           return <GifGridItem key={img.id} {...img} ></GifGridItem>
                        })
                }        
            
           {/*  <h3>{count}</h3>
            <button onClick={()=> setcount(count +1)}></button> */}
            </div>
        </div>
        
    )
}
