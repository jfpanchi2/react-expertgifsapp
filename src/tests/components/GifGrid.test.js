import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');
describe('Pruebas en GrifGrid', () => {
    const category = 'Cats';
    test("Debe de mostrar el componente correctamente", () => {
       useFetchGifs.mockReturnValue({
        data: [],
        loading: true,
       });
        const wrapper = shallow(<GifGrid categoria={category}></GifGrid>);
       expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs=[{
            id: 'ABC',
            url: 'https://localhost/imagen',
            title: 'Imagen gif'
        },
        {
            id: 'DEF',
            url: 'https://localhost/imagen',
            title: 'Imagen gif'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true,
        });
        const wrapper = shallow(<GifGrid categoria={category}></GifGrid>);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})
