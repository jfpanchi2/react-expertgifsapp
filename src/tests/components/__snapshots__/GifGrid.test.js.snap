// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GrifGrid Debe de mostrar el componente correctamente 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Cats
  </h3>
  <h4
    className="animate__animated animate__flash"
  >
    Cargando...
  </h4>
  <div
    className="card-grid"
  />
</div>
`;
